plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,403 @@
+                    .compileTemplate(TYPE, tag, newDocumentFieldName, newDocumentFieldName, scriptService);
+                ingestDocument.appendFieldValue(ingestField, ingestValue, false);
+            } else {
+                ArrayList<?> modelOutputValueArray;
+                if (modelOutputValue instanceof List) {
+                    List<?> modelOutputList = (List<?>) modelOutputValue;
+                    modelOutputValueArray = new ArrayList<>(modelOutputList);
+                } else if (modelOutputValue instanceof ArrayList) {
+                    modelOutputValueArray = (ArrayList<?>) modelOutputValue;
+                } else {
+                    throw new IllegalArgumentException("model output is not an array, cannot assign to array in documents.");
+                }
+
+                // check length of the prediction array to be the same of the document array
+                if (dotPathsInArray.size() != modelOutputValueArray.size()) {
+                    throw new RuntimeException(
+                        "the prediction field: "
+                            + originalModelOutputFieldName
+                            + " is an array in size of "
+                            + modelOutputValueArray.size()
+                            + " but the document field array from field "
+                            + newDocumentFieldName
+                            + " is in size of "
+                            + dotPathsInArray.size()
+                    );
+                }
+                // Iterate over dotPathInArray
+                for (int i = 0; i < dotPathsInArray.size(); i++) {
+                    String dotPathInArray = dotPathsInArray.get(i);
+                    Object modelOutputValueInArray = modelOutputValueArray.get(i);
+                    ValueSource ingestValue = ValueSource.wrap(modelOutputValueInArray, scriptService);
+                    TemplateScript.Factory ingestField = ConfigurationUtils
+                        .compileTemplate(TYPE, tag, dotPathInArray, dotPathInArray, scriptService);
+                    ingestDocument.appendFieldValue(ingestField, ingestValue, false);
+                }
+            }
+        } else {
+            throw new RuntimeException("model inference output cannot be null");
+        }
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static class Factory implements Processor.Factory {
+
+        private final ScriptService scriptService;
+        private final Client client;
+
+        public Factory(ScriptService scriptService, Client client) {
+            this.scriptService = scriptService;
+            this.client = client;
+        }
+
+        @Override
+        public MLInferenceIngestProcessor create(
+            Map<String, Processor.Factory> registry,
+            String processorTag,
+            String description,
+            Map<String, Object> config
+
```

Comment:
Remove this empty line

Reply:
changed

Reply:
(https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java#L395)